resources/user.go

```diff
@@ -224,6 +274,72 @@ type UserUID struct {
 	name string
 }
 
+// UserResAutoEdges holds the state of the auto edge generator.
+type UserResAutoEdges struct {
+	UIDs    []ResUID
+	pointer int
+}
+
+// AutoEdges returns edges from the user resource to each group found in
```

Comment:
Can you add a comment explaining the reverse stuff please? It would be very helpful. Thanks!